src/main/java/com/epam/controller/ReservationController.java

```diff
@@ -20,20 +20,23 @@ public ReservationController(ReservationService reservationService) {
     }
 
     @GetMapping("/testadmin")
-    public String listReservations(ModelMap modelMap) {
-        List<Reservation> list = reservationService.listReservations();
+    public String testadmin() {
+        return "redirect:/testadmin/1";
+    }
+
+    @GetMapping("/testadmin/{id}")
+    public String listReservations(@PathVariable int id, ModelMap modelMap) {
+        List<Reservation> list = reservationService.listReservations(id, reservationService.totalAmountOfRows);
+        int amount = reservationService.amountOfReservation();
         modelMap.addAttribute("listReservation", list);
+        modelMap.addAttribute("generalAmount",amount);
+        modelMap.addAttribute("amount", (amount / reservationService.totalAmountOfRows + 1));
         return "testadmin";
     }
 
-    @GetMapping("/deleteReservation/{id}")
+    @GetMapping("testadmin/deleteReservation/{id}")
```

Comment:
Let's extract @RequestMapping("/testamin") over the class

Reply:
done